sentry-core/src/session.rs

```diff
@@ -114,8 +114,109 @@ impl Session {
     }
 }
 
+// as defined here: https://develop.sentry.dev/sdk/envelopes/#size-limits
+const MAX_SESSION_ITEMS: usize = 100;
+const FLUSH_INTERVAL: u64 = 60;
+
+type SessionQueue = Arc<Mutex<Vec<SessionUpdate<'static>>>>;
+
+pub(crate) struct SessionFlusher {
```

Comment:
_nit_: Even though this is not public, it would be great to have doc comments explaining the high-level behavior of this type and its most important methods.